src/libs/utility/symbols.map

```diff
@@ -3,4 +3,24 @@ libelektra_0.8 {
 	elektraLskip;
 	elektraRstrip;
 	elektraStrip;
+};
+
+libelektra_1.0 {
+	elektraAbort;
+	elektraLog;
+	elektraCalloc;
+	elektraFormat;
+	elektraFree;
+	elektraMalloc;
+	elektraMemCaseCmp;
+	elektraMemDup;
+	elektraRealloc;
+	elektraStrCaseCmp;
+	elektraStrCmp;
```

Comment:
Maybe a bit off-topic, but why exactly do we need our own implementation of `strcmp`?

Reply:
The `symbols.map` files actually need to be cleanup when all this restructuring is done. `elektraStrCmp` isn't actually public anymore. It's now just an internal `static inline` function with extra asserts around standard `strcmp`.

https://github.com/ElektraInitiative/libelektra/blob/3af13a20f5fb48d6a2e6180add245c82da357f7c/src/include/internal/utility/compare.h#L40-L45

This PR actually turns most of these "just a standard libc function with extra asserts" into internal `static inline` functions. For the ones that need to be public (`elektraMalloc`, etc.) we re-export them via `libelektra-utility` as normal functions too, but internally use the `static inline` versions. More complicated things like `elektraFormat` are still normal functions everywhere, but are linked statically (but unexported) into `libelektra-core` to avoid a dependency.